src/collection.ts

```diff
@@ -1169,6 +1221,33 @@ export class Collection<
     )
   }
 
+  /**
+   * Retrieves one document from the KV store according to the given options.
+   *
+   * If no options are given, the first document in the collection is retreived.
+   * 
+   * @example
+   * ```ts
+   * // Get the first user
+   * const { result } = await db.users.getOne()
+   *
+   * // Get the first user with username that starts with "a"
+   * const { result } = await db.users.getOne({
+   *   filter: doc => doc.value.username.startsWith("a")
+   * })
+   * ```
+   *
+   * @param options - List options, optional.
+   * @returns A promise that resovles to the retreived document
+   */
+  async getOne(options?: ListOptions<Document<TOutput>>) {
+    const { result } = this.getMany({ ...options, limit: 1 })
```

Comment:
limit: 1
Here causes only the first document to be taken from the database, which means if a filter is provided, it is quite unlikely to find a match since it is only checking the first document it comes across.

I suggest using the internal collection.handleMany() method, which I have now modified to enable all handling of one or limited results. Pass it the options { ...options, resultLimit: 1 }, and then you can simply: return result.at(0) ?? null